# 1141_社會資料分析 作業七
學生：劉晏成

以「2018台北大學社會系三鶯調查」（brilliant_working.sav）檔案回答以下問題：

1. 以收入（inc）作為依變項，性別（sex）為自變項，檢驗教育年數（eduy）是否可以當做上述二者的中介變項。  
如果有中介關係，計算 (bc - b‘c) / bc ，顯示中介關係占原關係的百分比
2. 以收入（inc）作為依變項，教育年數（eduy）做為自變項、人際網路（net）作為調節變項，檢驗教育年數對收入的影響是否受到人際網路調節。

## Notes
1. 請先做整體模型的描述(R平方、顯著性等)。
2. 第一題請列出迴歸方程式與計算過程。若無需繼續檢視中介效果請以文字說明(請參考本週簡報p.27)。
3. 第二題交互作用項顯著與否的後續作法不同，請操考調節變項的示範一與練習一，但不需附上折線圖。
4. 請看清楚自變項、依變項、中介、調節分別是哪個變項，也請看清楚P值以免做出錯誤判斷。

In [1]:
from load import load_sav
import pandas as pd

df = load_sav("../data/brilliant_working.sav")

In [2]:
variable_value_labels = df.attrs['variable_value_labels']
column_names = df.attrs['column_names']
column_names_to_labels = df.attrs['column_names_to_labels']

## Q1: 以收入（inc）作為依變項，性別（sex）為自變項，檢驗教育年數（eduy）是否可以當做上述二者的中介變項。
如果有中介關係，計算 (bc - b‘c) / bc ，顯示中介關係占原關係的百分比

In [3]:
df_1 = df[['inc', 'sex', 'eduy']].copy()

檢查遺漏值狀況

In [4]:
from statistic.missing_value import check_missing_value

check_missing_value(df_1)

# of Missing Values
inc                     4
sex                     0
eduy                    0

發現有遺漏值，處理遺漏值

In [5]:
df_1_cleared = df_1.dropna()
check_missing_value(df_1_cleared)

# of Missing Values
inc                     0
sex                     0
eduy                    0

In [7]:
from statistic.linear_regression import mediation_analysis
reports = mediation_analysis(X = df_1_cleared['sex'], Y = df_1_cleared['inc'], Z = df_1_cleared['eduy'])

檢視 $Y = b_cX + a_1 $ 迴歸結果

In [8]:
reports['total_effect_summary']

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    inc   R-squared:                       0.048
Model:                            OLS   Adj. R-squared:                  0.047
Method:                 Least Squares   F-statistic:                     43.67
Date:                Wed, 10 Dec 2025   Prob (F-statistic):           6.81e-11
Time:                        23:43:26   Log-Likelihood:                -10119.
No. Observations:                 864   AIC:                         2.024e+04
Df Residuals:                     862   BIC:                         2.025e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       2.992e+04   1396.482     21.427      0.000    2.72e+04    3.27e+04
sex          1.33e+04   2012.545      6.608      0.000    9349.217    1.72e+04
==============================================================================
Omnibus:                      217.946   Durbin-Watson:                   1.964
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              499.360
Skew:                           1.357   Prob(JB):                    3.68e-109
Kurtosis:                       5.551   Cond. No.                         2.58
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

小結：$b_c$ 為13300， P value 為0；由於P value < 0.05 ，具統計顯著性

檢視 $Z = b_aX + a_2 $ 迴歸結果

In [10]:
reports['mediator_summary']

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   eduy   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                 -0.000
Method:                 Least Squares   F-statistic:                    0.8234
Date:                Wed, 10 Dec 2025   Prob (F-statistic):              0.364
Time:                        23:43:26   Log-Likelihood:                -2367.4
No. Observations:                 864   AIC:                             4739.
Df Residuals:                     862   BIC:                             4748.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         13.4509      0.177     75.886      0.000      13.103      13.799
sex            0.2318      0.255      0.907      0.364      -0.270       0.733
==============================================================================
Omnibus:                       30.342   Durbin-Watson:                   1.773
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               32.551
Skew:                          -0.464   Prob(JB):                     8.54e-08
Kurtosis:                       2.791   Cond. No.                         2.58
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

小結：$b_a$ 為 0.2318， P value 為0.364；由於P value > 0.05 ，不具統計顯著性

### Answer
整體模型的R Square 為0.048，具統計顯著性。

由於 $b_a$ 不具統計顯著性，中介檢驗不通過，停止檢驗。教育年份對收入與性別無中介作用。